### PR TITLE
UCP/ADDRESS: Pack interface capability bits to reduce adress size.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by yosefe** [comment]
@MattBBaker can you please check if this makes address size small enough so rndv works?
i did not enable rndv on ugni with this PR because there may be other problems with rndv on ugni

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1637/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2312/ for details.

**Comment by MattBBaker** [comment]
@yosefe Yes, this patch fixes the wireup problem.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2313/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1638/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@MattBBaker does rndv work on ugni now (i guess removed the hack which disables it)?
@shamisp can you pls review as well?

File: src/ucp/wireup/address.c [hunk 1]
```diff
@@ -44,6 +44,13 @@ typedef struct {
...
 
+typedef struct {
+    float            overhead;
+    float            bandwidth;
+    uint32_t         prio_cap_flags; /* 8 lsb: prio, 24 msb - cap flags */
+} ucp_address_packed_iface_attr_t;
+
+
 #define UCP_ADDRESS_FLAG_LAST         0x80   /* Last address in the list */
 #define UCP_ADDRESS_FLAG_EMPTY        0x80   /* Device without TL addresses */
```
Comment (shamisp) [inline src/ucp/wireup/address.c]: not related to this patch, but it is ok that both defines 0x80

**Comment by shamisp** [comment]
Awesome 👍  @MattBBaker - please confirm.

File: src/ucp/wireup/address.c (hunk 1 above)
Comment (yosefe) [inline src/ucp/wireup/address.c]: they are used in different parts in the packed address and i didn't want to force them to be the same

**Comment by raffenet** [comment]
This looks like it will alleviate some of the address length issues we were having during address exchange in `MPI_Init`. :+1:

**Comment by MattBBaker** [comment]
Just tested, runs full test fine with only known expected failures. :+1:

**Comment by yosefe** [comment]
@MattBBaker which expected failures we have?

**Comment by MattBBaker** [comment]
@yosefe I haven't implemented the non blocking flush callback yet.

**Pull request merged**